app/src/main/java/com/gh4a/activities/EditCommitCommentActivity.java

```diff
@@ -4,35 +4,42 @@
 import android.content.Intent;
 
 import com.gh4a.Gh4Application;
+import com.gh4a.utils.ApiHelpers;
+import com.meisolsson.githubsdk.model.GitHubCommentBase;
+import com.meisolsson.githubsdk.model.request.CommentRequest;
+import com.meisolsson.githubsdk.model.request.repository.CreateCommitComment;
+import com.meisolsson.githubsdk.service.repositories.RepositoryCommentService;
 
-import org.eclipse.egit.github.core.CommitComment;
-import org.eclipse.egit.github.core.RepositoryId;
-import org.eclipse.egit.github.core.service.CommitService;
-
-import java.io.IOException;
+import io.reactivex.Single;
 
 public class EditCommitCommentActivity extends EditCommentActivity {
     public static Intent makeIntent(Context context, String repoOwner, String repoName,
-            String commitSha, CommitComment comment) {
+            String commitSha, long id, String body) {
         Intent intent = new Intent(context, EditCommitCommentActivity.class)
                 .putExtra("commit", commitSha);
-        return EditCommentActivity.fillInIntent(intent,
-                repoOwner, repoName, comment.getId(), 0L, comment.getBody(), 0);
+        return EditCommentActivity.fillInIntent(intent, repoOwner, repoName, id, 0L, body, 0);
     }
 
     @Override
-    protected void createComment(RepositoryId repoId, CommitComment comment,
-            long replyToCommentId) throws IOException {
-        Gh4Application app = Gh4Application.get();
-        CommitService commitService = (CommitService) app.getService(Gh4Application.COMMIT_SERVICE);
+    protected Single<GitHubCommentBase> createComment(String repoOwner, String repoName,
+            String body, long replyToCommentId) {
+        RepositoryCommentService service =
+                Gh4Application.get().getGitHubService(RepositoryCommentService.class);
+        CreateCommitComment request = CreateCommitComment.builder().body(body).build();
         String sha = getIntent().getStringExtra("commit");
-        commitService.addComment(repoId, sha, comment);
+        return service.createCommitComment(repoOwner, repoName, sha, request)
+                .map(ApiHelpers::throwOnFailure)
+                .map(response -> response);
```

Comment:
What's the point of this map?

Reply:
Making the type system happy ;-)
Here: convert from GitComment to GitHubCommentBase.

Reply:
And what makes it unhappy? If I remove that line I don't see any warnings and the app still compiles.

Reply:
It didn't compile when I introduced this. It said that a Single<GitComment> isn't compatible to the supposed return type Single<GitHubCommentBase>.

I'll check this again.